Move away from @cray.com email addresses

bradcray opened the pull request:
This PR updates the code base w.r.t. @cray.com email addresses as follows:

* removed files that contained email addresses yet don't seem all that
  important to the current workflow or software release.  Specifically:
  - GOALS.md (I think we're now far enough along not to need to advertise
    our goals apart from "take over the world")
  - future_stats (I don't think we use this anymore)
  - STATUS.devel (at this point, I think we're pretty invested in GitHub issues
    and wouldn't really encourage users to refer to this list anymore)
  - lookForTabs.cron (we just use lookForTabs now)

  Also updated things referring/relating to the above files, such as:
  - Makefile.devel
  - README.devel
  - README.files
  - FUTURES-semantic
  - gen_release
  - install.sh

* removed plaintext email addresses from docs.  Even though they had :disguise:
  tags on them, the fact that the .rst sources are in our repo and text sources on
  the website didn't make this as effective as intended.
  - cygwin.rst
  - extern.rst
  - firstClassFns.rst
  - building.rst
  - chplenv.rst
  - tasks.rst
  - chpl-mode.el

* removed plaintext emails from code as well:
  - */sparse.chpl
  - util/build_configs/cray-internal/generate-dev-releaseinfo.bash

* removed any mention of supporting a private bugs mailing list.  I don't think we've
  gotten enough activity on the list to warrant creating a replacement for it, and can
  consider doing so if/when the need arises.
  - bugs.rst
  - privatebugs.rst

* updated scripts/docs that try to auto-infer a user's email address and tack it onto a
  cray.com address in debug mode:
  - testRelease
  - README
  - nightly
  - tokctnightly

* updated `${USER}@cray.com` with `${USER}@hpe.com` in `start_opt_test` which is
  not currently in-use, but a bit entangled in other scripts and mentioned as something
  we'd like to re-enable.  Likely this will need more work before it's useful/usable again,
  or removed altogether.

* removed some text files of email conversations which I can't
  recall why I would've intentionally put into the repository.
  - FEEDBACK*

bradcray commented on util/cron/nightly:
```diff
@@ -248,7 +248,7 @@ $user = `whoami`;
 $debugmail = $ENV{'CHPL_NIGHTLY_DEBUG_EMAIL'};
 if ($debugmail eq "") {
-    $debugmail = "$user\@cray.com";
+    $debugmail = "$user\@hpe.com";
```
Yeah, I realized this style of change was unlikely to "just work" but also seemed unlikely to be used much.  Do you have other thoughts about how to address this short of rewriting the script to accept a new environment variable or argument or something like that?  I suppose it might be best to just put an error message into this conditional?

ronawho replied:
Yeah, I might just make it an error if `$debug == 1` and `CHPL_NIGHTLY_DEBUG_EMAIL` isn't set

bradcray replied:
I did this for this and tokctnightly.

bradcray commented on util/cron/start_opt_test:
```diff
@@ -75,7 +75,7 @@ foreach $line (@lines) {
 if ($cron && (!exists($ENV{"CHPL_TEST_NOMAIL"}) or grep {$ENV{"CHPL_TEST_NOMAIL"} =~ /^$_$/i} ('','\s*','0','f(alse)?','no?'))) {
```
I don't have any real memory of this one.  Are we definitely not using it anymore?

ronawho replied:
We are definitely not (see util/cron/README) but there are a few other scripts that reference it, so maybe leave it in for now, and we can remove it in a follow up PR.

ronawho replied:
(looks like we check that this script has no syntax errors, and that's what causing the travis failure. I think we can remove all references to it, but for this PR it probably makes sense to just re-add it)

bradcray commented on highlight/emacs/chpl-mode.el:
```diff
@@ -1,7 +1,7 @@
 ;;; chpl-mode.el --- Major mode for editing Chapel code
 
 ;; Author:     2007 Steven T Balensiefer
-;; Maintainer: Chapel group <chapel_info@cray.com>
```
I took the general policy of not removing personal @cray.com email addresses in cases where the author put them in given that they haven't been active for years anyway and are part of the historical record (i.e., I wasn't trying to scrub all @cray.com addresses, just looking for ones that needed to be updated and then doing some cleanup for cases that didn't really make sense anymore.

bradcray replied:
(arguably I could've left this one in by the same token, but...).  Plus, it looks like we should be getting a new / good emacs mode soon, so this file may go away soon anyway (fingers crossed).

ronawho replied:
Sounds good

bradcray commented on GOALS.md:
```diff
@@ -1,28 +0,0 @@
-Chapel Public Release Goals
```
Wow, I'm not sure how I missed those...

bradcray commented on Makefile.devel:
```diff
@@ -64,10 +64,3 @@ SPECTEST_DIR = ./test/release/examples/spec
-
-
-future_stats: FUTURES
-	./util/devel/test/future_stats STATUS.devel test/FUTURES
```
OK, done.

bradcray commented on util/tokencount/tokctnightly:
```diff
@@ -47,7 +47,7 @@ $user = `whoami`;
 chomp($user);
```
I ended up leaving this in because I either wanted to blow away this whole directory or none of it, but didn't want to take the time to review what's in it right now.

ronawho replied:
Yeah, makes sense.

bradcray commented on Makefile.devel:
```diff
@@ -64,10 +64,3 @@ SPECTEST_DIR = ./test/release/examples/spec
-
-
-future_stats: FUTURES
-	./util/devel/test/future_stats STATUS.devel test/FUTURES
```
I agree, thanks.

bradcray commented:
I think I've addressed all your comments.  I also updated the OP to be more complete than it originally was (I meant to edit it further before asking for a review, but apparently didn't).

bradcray commented:
OK, I've restored `start_opt_test`, and am ready to merge if you are.
